lib/modules/platform/scm-manager/index.ts

```diff
@@ -0,0 +1,338 @@
+  return repos.map((repo) => `${repo.namespace}/${repo.name}`);
+}
+
+export async function getBranchPr(branchName: string): Promise<Pr | null> {
+  return await findPr({ branchName, state: 'open' });
+}
+
+export async function findPr({
+  branchName,
+  prTitle,
+  state = 'all',
+}: FindPRConfig): Promise<Pr | null> {
+  const inProgressPrs = await getPrList();
+  const result = inProgressPrs.find(
+    (pr) =>
+      branchName === pr.sourceBranch &&
+      (!prTitle || prTitle === pr.title) &&
+      matchPrState(pr, state),
+  );
+
+  if (result) {
+    logger.trace({ result }, `Found PR`);
+    return result;
+  }
+
+  logger.trace(
+    `Could not find PR with source branch ${branchName} and title ${
+      prTitle ?? ''
+    } and state ${state}`,
+  );
+
+  return null;
+}
+
+export async function getPr(number: number): Promise<Pr | null> {
+  const inProgressPrs = await getPrList();
+  const cachedPr = inProgressPrs.find((pr) => pr.number === number);
+
+  if (cachedPr) {
+    logger.trace('Returning from cached PRs');
+    return cachedPr;
+  }
+
+  try {
+    const result = await getRepoPr(config.repository, number);
+    logger.trace('Returning PR from API');
+    return mapPrFromScmToRenovate(result);
+  } catch (error) {
+    logger.error({ error }, `Can not find a PR with id ${number}`);
+    return null;
+  }
+}
+
+export async function getPrList(): Promise<Pr[]> {
+  if (config.prList === null) {
+    try {
+      config.prList = (await getAllRepoPrs(config.repository)).map((pr) =>
+        mapPrFromScmToRenovate(pr),
+      );
+    } catch (error) {
+      logger.error(error);
+    }
+  }
+
+  return config.prList ?? [];
+}
+
+export async function createPr({
+  sourceBranch,
+  targetBranch,
+  prTitle,
+  prBody,
+  draftPR,
+}: CreatePRConfig): Promise<Pr> {
+  const createdPr = await createScmPr(config.repository, {
+    source: sourceBranch,
+    target: targetBranch,
+    title: prTitle,
+    description: sanitize(prBody),
+    status: draftPR ? 'DRAFT' : 'OPEN',
+  });
+
+  logger.info(
+    `PR created with title '${createdPr.title}' from source '${createdPr.source}' to target '${createdPr.target}'`,
+  );
+
+  return mapPrFromScmToRenovate(createdPr);
+}
+
+export async function updatePr({
+  number,
+  prTitle,
+  prBody,
+  state,
+  targetBranch,
+}: UpdatePrConfig): Promise<void> {
+  await updateScmPr(config.repository, number, {
+    title: prTitle,
+    description: sanitize(prBody) ?? undefined,
+    target: targetBranch,
+    status: mapPrState(state),
+  });
+
+  logger.info(`Updated PR #${number} with title ${prTitle}`);
+}
+
+export function mergePr(config: MergePRConfig): Promise<boolean> {
+  logger.debug('Not implemented mergePr');
+  return Promise.resolve(false);
+}
+
+export function getBranchStatus(
+  branchName: string,
+  internalChecksAsSuccess: boolean,
+): Promise<BranchStatus> {
+  logger.debug('Not implemented getBranchStatus');
+  return Promise.resolve('red');
```

Comment:
note: no automerge possible without this

Reply:
Can be done later.